tests/upgrades/test_provisioningtemplate.py

```diff
@@ -0,0 +1,141 @@
+        """
+        host = module_target_sat.api.Host(
+            organization=module_org,
+            location=module_location,
+            name=gen_string('alpha'),
+            operatingsystem=default_os,
+            architecture=default_architecture,
+            domain=default_domain,
+            root_pass=settings.provisioning.host_root_password,
+            ptable=default_partitiontable,
+            pxe_loader=pxe_loader.pxe_loader,
+        ).create()
+
+        for kind in provisioning_template_kinds:
+            assert host.read_template(data={'template_kind': kind})
+
+        save_test_data(
+            {
+                'provision_host_id': host.id,
+                'pxe_loader': pxe_loader.pxe_loader,
+            }
+        )
+
+    @pytest.mark.post_upgrade(depend_on=test_pre_scenario_provisioning_templates)
+    @pytest.mark.parametrize('pre_upgrade_data', ['bios', 'uefi'], indirect=True)
+    def test_post_scenario_provisioning_templates(
+        self,
+        request,
+        pre_upgrade_data,
+        module_target_sat,
+    ):
+        """Host provisioned using pre-upgrade GCE CR
+
+        :id: postupgrade-ef82143d-efef-49b2-9702-93d67ef6805e
+
+        :steps:
+            1. Postupgrade, verify provisioning templates rendering for host
+            2. Create a new host on Satellite and try rendering provisioning templates
+
+        :expectedresults:
+            1. Provisioning templates for existing and new host are able to render.
+
+        :parametrized: yes
+        """
+        pxe_loader = pre_upgrade_data.pxe_loader
+        pre_upgrade_host = module_target_sat.api.Host().search(
+            query={'search': f'id={pre_upgrade_data.provision_host_id}'}
+        )[0]
+        org = module_target_sat.api.Organization(id=pre_upgrade_host.organization.id).read()
+        loc = module_target_sat.api.Location(id=pre_upgrade_host.location.id).read()
+        domain = module_target_sat.api.Domain(id=pre_upgrade_host.domain.id).read()
+        architecture = module_target_sat.api.Architecture(
+            id=pre_upgrade_host.architecture.id
+        ).read()
+        os = module_target_sat.api.OperatingSystem(id=pre_upgrade_host.operatingsystem.id).read()
+        ptable = module_target_sat.api.PartitionTable(id=pre_upgrade_host.ptable.id).read()
+
+        for kind in provisioning_template_kinds:
+            assert pre_upgrade_host.read_template(data={'template_kind': kind})
+
+        new_host_name = gen_string('alpha')
+        new_host = module_target_sat.api.Host(
+            name=new_host_name,
+            organization=org,
+            location=loc,
+            architecture=architecture,
+            domain=domain,
+            operatingsystem=os,
+            ptable=ptable,
+            root_pass=settings.provisioning.host_root_password,
+            pxe_loader=pxe_loader,
+        ).create()
+        request.addfinalizer(pre_upgrade_host.delete)
+        request.addfinalizer(new_host.delete)
```

Comment:
non-blocking. Since there's more than one line for teardown you could use 
```python
@request.addfinalizer
    def _finalize():
```

Reply:
yes, I didn't use finalize method here intensionally, as it was simple teardown with 2 lines, and we don't use extra unnecessary lines of code here for method definition